# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1346986925 14400
#      Thu Sep 06 23:02:05 2012 -0400
# Node ID c42bf5cadb604308770b13bc891732324d135554
# Parent  9f9a7ae2d263c16cd4daf6cb01dc0cfdbcd68392
Add check for CURLOPT_DIRLISTONLY in libcurl

* configure.ac: Add check for CURLOPT_DIRLISTONLY in libcurl

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -834,16 +834,30 @@ CPPFLAGS="$save_CPPFLAGS"
 
 save_CPPFLAGS="$CPPFLAGS"
 save_LIBS="$LIBS"
 CPPFLAGS="$Z_CPPFLAGS $CPPFLAGS"
 LIBS="$Z_LDFLAGS $Z_LIBS $LIBS"
 OCTAVE_CHECK_LIB(curl, cURL,
   [cURL library not found.  The ftp objects, urlread and urlwrite functions will be disabled.],
   [curl/curl.h], [curl_easy_escape])
+if test -z "$warn_curl"; then
+  AC_MSG_CHECKING([for CURLOPT_DIRLISTONLY in curl/curl.h])
+  AC_COMPILE_IFELSE(
+    [AC_LANG_PROGRAM([[
+      #include <curl/curl.h>
+      ]], [[
+      curl_easy_setopt ((CURL*)NULL, CURLOPT_DIRLISTONLY, 0);
+      ]])],
+    [AC_MSG_RESULT(yes)],
+    [AC_MSG_RESULT(no)
+     AC_DEFINE(CURLOPT_DIRLISTONLY, CURLOPT_FTPLISTONLY,
+       [Define to the legacy option name if using an older version of cURL.])]
+  )
+fi
 LIBS="$save_LIBS"
 CPPFLAGS="$save_CPPFLAGS"
 
 ### Check for either of Graphics/ImageMagick++ libraries
 
 AC_ARG_WITH([magick],
   [AS_HELP_STRING([--with-magick=LIB],
     [select library to use for image I/O (options: GraphicsMagick(default) or ImageMagick)])],
